be_repo/app.py

```diff
@@ -18,21 +35,62 @@
     raise Exception("Unable to find the document due to the following error: ", e)
 
 
-@app.route('/upload', methods=['POST'])
-def upload_resume():
+@app.route('/upload', methods=['POST', 'OPTIONS'])
+def upload_resume(): 
+    if request.method == 'OPTIONS':
+        # Allows the preflight request to succeed
+        return jsonify({'status': 'OK'}), 200
+   
     user_id = request.form.get('user_id')
     if not user_id:
         return jsonify({"error": "No user ID provided."}), 400
     # Vector the resume text
     return upload_parse_resume(request, resume_collection)
 
-
-@app.route('/resume_evaluate', methods=['POST'])
+@app.route('/login', methods=['POST', 'OPTIONS'])
+def login():
+    if request.method == 'OPTIONS':
+        # Allows the preflight request to succeed
+        return jsonify({'status': 'OK'}), 200
+    
+    token = request.form.get('access_token')
+    if not token:
+        return jsonify({'status': 'error', 'message': 'ID token is missing'}), 400
+    
+    try:
+        # Verify the token with Google
+        idinfo = id_token.verify_oauth2_token(token, google_requests.Request(), GOOGLE_CLIENT_ID)
+        
+        if idinfo['aud'] != GOOGLE_CLIENT_ID:
+            return jsonify({'status': 'error', 'message': 'Token audience mismatch'}), 400
+
+        # Token is valid, extract user information
+        userid = idinfo['sub']
+        email = idinfo['email']
+        name = idinfo.get('name', 'No name available')
+
+        # Create a session or JWT for the user
+        session.permanent = True 
+        session['user'] = {'userid': userid, 'email': email, 'name': name}
+
+        # Respond with success and optional user data
+        return jsonify({'status': 'success', 'email': email, 'name': name}), 200
+ 
+    except ValueError as e:
+        # Token verification failed
+        print(f"Error verifying token: {str(e)}")
+        return jsonify({'status': 'error', 'message': 'Invalid ID token'}), 400
+
+@app.route('/resume_evaluate', methods=['POST', 'OPTIONS'])
```

Comment:
From my view, the login has nothing to do with E2E test. You'd better to open two PRs for them. @andyasdd1